pkg/operator/ceph/cluster/operator_watchers.go

```diff
@@ -44,7 +44,10 @@ func (c *ClusterController) StartOperatorSettingsWatch(namespace string, stopCh
 // StopWatch stop watchers
 func (c *ClusterController) StopWatch() {
 	for _, cluster := range c.clusterMap {
-		close(cluster.stopCh)
+		// check channel is open before closing
+		if _, ok := <-cluster.stopCh; ok {
+			close(cluster.stopCh)
```

Comment:
Where else is the channel being closed? When it's closed it should be removed from the `clusterMap`, rather than trying to read from it. Reading from the channel actually won't be a good way to check if it's closed since if it is not closed, it will block until it is written to.

Reply:
@travisn currently, its getting closed at https://github.com/rook/rook/blob/9c23522f2bc0e19facdf7cb30524d9549565c402/pkg/operator/ceph/cluster/controller.go#L376 and https://github.com/rook/rook/blob/9c23522f2bc0e19facdf7cb30524d9549565c402/pkg/operator/ceph/cluster/operator_watchers.go#L47

Reply:
So this one should probably be removed https://github.com/rook/rook/blob/9c23522f2bc0e19facdf7cb30524d9549565c402/pkg/operator/ceph/cluster/controller.go#L376

Reply:
Where it is stopped in controller.go, it also sets `closedStopCh = true`. Can we just check that flag and skip this close?

Reply:
changed the validation based on the closedStopCh flag